### PR TITLE
apd: add go 1.21 and 1.22 testing to CI

### PR DESCRIPTION
**Pull request by nvanbenschoten**
This commit adds a test variant for go 1.21 and 1.22.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -71,9 +73,9 @@ jobs:
       - name: 'GCAssert'
         # Only run gcassert on the latest versions of Go. Inlining heuristics
         # change from version to version.
```
Comment (petermattis) [inline .github/workflows/go.yml:75]: This comment seems out of date. We run gcassert on go1.18+.

**Review by petermattis — APPROVED** [review]
LGTM

Did you consider removing testing for go1.13 and go1.14? Not sure if it matters. This repo rarely changes.

File: .github/workflows/go.yml (hunk 1 above)
Comment (nvanbenschoten) [reply .github/workflows/go.yml:75]: "the latest versions" was meant to imply that there might be a few recent versions on which we run gcassert, not just the single latest.

**Comment by nvanbenschoten** [comment]
> Did you consider removing testing for go1.13 and go1.14? Not sure if it matters. This repo rarely changes.

Yeah, we should go through and remove old versions from CI, now that they're quite stale. I'll leave that for next time, as they're also not hurting anything.

**Pull request merged**